### PR TITLE
[SPARK-38938][PYTHON] Implement `inplace` and `columns` parameters of `Series.drop`

### PR DESCRIPTION
**Pull request by xinrong-meng**
### What changes were proposed in this pull request?
Implement `inplace` and `columns` parameters of `Series.drop`.

### Why are the changes needed?
Increase pandas API coverage.

### Does this PR introduce _any_ user-facing change?
Yes. `inplace` and `columns` parameters of `Series.drop` are supported now.

```py
>>> s = ps.Series(data=np.arange(3), index=['A', 'B', 'C'])

# 'columns' parameter
>>> s.drop(columns=['A'])
A    0                                                                          
B    1
C    2
dtype: int64

# 'inplace' parameter
>>> s.drop(index=['B', 'C'], inplace=True)
>>> s
A    0
dtype: int64
```

### How was this patch tested?
Unit tests.

### CODE REVIEW

**Comment by xinrong-databricks** [comment]
Please refer to https://github.com/xinrong-databricks/spark/actions/runs/2185003953 for test results.
@ueshin @HyukjinKwon @itholic Thanks!

**Comment by ueshin** [comment]
When `inplace=True`, it shouldn't update the anchor:

```py
>>> pdf = pd.DataFrame({"a": np.arange(3)}, index=['A', 'B', 'C'])
>>> pdf
   a
A  0
B  1
C  2
>>> s = pdf.a
>>> s.drop(index=['B', 'C'], inplace=True)
>>> s
A    0
Name: a, dtype: int64
>>> pdf
   a
A  0
B  1
C  2
```

whereas:

```py
>>> psdf = ps.DataFrame({"a": np.arange(3)}, index=['A', 'B', 'C'])
>>> ss = psdf.a
>>> ss.drop(index=['B', 'C'], inplace=True)
>>> ss
A    0
Name: a, dtype: int64
>>> psdf
   a
A  0
```

**Comment by xinrong-databricks** [comment]
Thanks @ueshin , good catch! Fixed.

**Comment by xinrong-databricks** [comment]
Please refer to https://github.com/xinrong-databricks/spark/actions/runs/2186250376 for test results.
@ueshin @HyukjinKwon @itholic Thanks!

**Comment by xinrong-databricks** [comment]
Rebased for CI.

**Review by itholic — APPROVED** [review]
Looks fine to me

File: python/pyspark/pandas/tests/test_series.py [hunk 1]
```diff
@@ -1749,14 +1759,22 @@ def test_drop(self):
...
+            psser.drop("lama", columns="cow")
+
         msg = r"'Key length \(2\) exceeds index depth \(3\)'"
         with self.assertRaisesRegex(KeyError, msg):
             psser.drop(("lama", "speed", "x"))
 
+        psser.drop({"lama": "speed"}, inplace=True)
+        pser.drop({"lama": "speed"}, inplace=True)
+        self.assert_eq(psser, pser)
+        self.assert_eq(psdf, pdf)
```
Comment (itholic) [inline python/pyspark/pandas/tests/test_series.py:1776]: Oh, can we add a one test for chained case ?

e.g.
```python
        (psser + 1).drop({"lama": "speed"}, inplace=True)
        (pser + 1).drop({"lama": "speed"}, inplace=True)
        self.assert_eq(psser, pser)
        self.assert_eq(psdf, pdf)
```

Comment (xinrong-databricks) [reply python/pyspark/pandas/tests/test_series.py:1781]: Added, thanks!

File: python/pyspark/pandas/series.py [hunk 2]
```diff
@@ -2377,10 +2379,18 @@ def drop(
         ----------
         labels : single label or list-like
             Index labels to drop.
-        index : None
+        index : single label or list-like
```
Comment (itholic) [inline python/pyspark/pandas/series.py:2382]: +1

**Review by itholic — APPROVED** [review]
Looks pretty good!

**Comment by xinrong-databricks** [comment]
Rebased to update `supported_pandas_api.rst`.

**Review by HyukjinKwon — APPROVED** [review]

**Comment by HyukjinKwon** [comment]
Merged to master.

**Pull request closed without merging**